src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -0,0 +1,84 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.sps.data.HuntItem;
+import com.google.sps.data.Riddle;
+import com.google.sps.data.ScavengerHunt;
+import java.io.IOException;
+import java.util.ArrayList;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/go-data")
+public class DataServlet extends HttpServlet {
+  private static final String JSON_TYPE = "application/json";
+
+  private ScavengerHunt hunt;
+  private int index = -1;
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String indexStr = request.getParameter("new-index");
+    index = Integer.parseInt(indexStr);
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    buildScavengerHunt();
+
+    response.setContentType(JSON_TYPE);
+    Gson gson = new Gson();
+    String json = gson.toJson(hunt);
```

Comment:
I'm wondering if the hunt variable here needs to be a member field in this class.  It appears that buildScavengerHunt() is building object and setting the member field, which has some side effects that aren't obvious.

I would recommend just having  buildScavengerHunt return the ScavengerHunt object instead of setting a class variable like this.